### PR TITLE
feat(metric-engine): open and close metric regions

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Handle open and close region requests for metric engine.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
Related to #2070

### CODE REVIEW

File: src/metric-engine/src/engine/close.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    ) -> Result<AffectedRows> {
+        let data_region_id = utils::to_data_region_id(region_id);
+        if self
+            .state
+            .read()
+            .await
+            .physical_regions()
+            .contains_key(&data_region_id)
+        {
+            self.close_physical_region(data_region_id).await?;
```
Comment (killme2008) [inline src/metric-engine/src/engine/close.rs:50]: I think it's better to add some info logs when opening or closing regions.

File: src/metric-engine/src/metadata_region.rs [hunk 2]
```diff
@@ -157,6 +157,24 @@ impl MetadataRegion {
 
         Ok(columns)
     }
+
+    pub async fn logical_regions(&self, physical_region_id: RegionId) -> Result<Vec<RegionId>> {
+        let metadata_region_id = utils::to_metadata_region_id(physical_region_id);
+        let region_prefix = "__region_".to_string();
```
Comment (killme2008) [inline src/metric-engine/src/metadata_region.rs:163]: It's also used in `src/metric-engine/src/metadata_region.rs`.

I think we can extract a constant.

**Review by killme2008 — APPROVED** [review]
Almost LGTM

File: src/metric-engine/src/engine.rs [hunk 3]
```diff
@@ -157,11 +159,14 @@ impl RegionEngine for MetricEngine {
...
     async fn stop(&self) -> Result<(), BoxedError> {
-        todo!()
+        // don't need to stop the underlying mito engine
+        Ok(())
     }
 
     fn set_writable(&self, region_id: RegionId, writable: bool) -> Result<(), BoxedError> {
-        todo!()
+        // ignore the region not found error
+        let _ = self.inner.mito.set_writable(region_id, writable);
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/metric-engine/src/engine.rs:168]: This is mainly for logical region.

[the remark this answers is not in the document]
Comment (evenyag) [reply src/metric-engine/src/engine.rs:168]: If the region doesn't exist, we don't need to invoke `set_writable()`.

[the remark this answers is not in the document]
Comment (waynexia) [reply src/metric-engine/src/engine.rs:168]: I cannot know if it exists without `.await`. WDYT of only ignoring `RegionNotFound`?

[the remark this answers is not in the document]
Comment (evenyag) [reply src/metric-engine/src/engine.rs:168]: Got it. We can only ignore `RegionNotFound`. I remember that there is a status code for it.

**Review by evenyag — APPROVED** [review]
Rest LGTM.

**Pull request merged**